### ISSUE TITLE
Consul grpc_tls usage cannot deploy nomad jobs anymore

### ISSUE THREAD

**Issue by suikast42** [hashicorp/nomad#15266]
### Nomad version
1.4.2
### Consul version
1.14.0

I updated my consul agent today. Consul 1.40 have a breaking change 

BREAKING CHANGES:
config: Add new ports.grpc_tls configuration option.

I remove the grpc falg form the config and replace it with grpc_tls . After that nomad is not able to deploy jobs with the error:

Placement Failure
Task Group "keycloak":
  * Constraint "${attr.consul.grpc} > 0": 1 nodes excluded by filter

**Comment by jrasell** [comment]
Hi @suikast42 and thanks for raising this issue. We are currently looking into how this change impacts Nomad's integration, and what, if any fixes would be needed.

**Comment by suikast42** [comment]
Regarding to your comment https://github.com/hashicorp/nomad/issues/15295#issuecomment-1319681122



The working consul 1.13.3 sever and agent config 

Consul 1.14.0 does not booting up if the grpc_tls  is missing. So I add only 
grpc_tls = 8503 to the  ports section. 
If I delete the grpc sec falg in tthe ports section then nomad can't deploy artifacts because of missinf not attribute grpc. 

I hope the was the right answer. 

```json
{
    "node_name": "{{host_name}}",
    "datacenter": "{{data_center}}",
    "data_dir": "{{consul_data_dir}}",
    "server": true,
    "log_level" : "INFO",
    "bind_addr": "0.0.0.0",
    "advertise_addr": "{{host_ip}}",
    "client_addr": "0.0.0.0",
    "encrypt": "{{consul_encrypt_key}}",
    "ui_config": {
        "enabled" : true
    },
    "addresses": {
        "grpc" : "127.0.0.1"
    },
    "ports": {
        "http": -1,
        "grpc" : 8502,
        "https": 8501
    },
    "connect": {
        "enabled": true
    },
    "retry_join":{{masters | to_json }},
    "bootstrap_expect": {{masters|length}},

    "auto_encrypt": {
        "allow_tls": true
    },

	"performance" :{
	   "raft_multiplier" : 1
	},

    "node_meta": {
        "node_type": "server"
    },
    "tls":{
      "defaults":{
        "ca_file": "{{cluster_intermediate_ca_bundle}}",
        "cert_file": "{{consul_cert}}",
        "key_file": "{{consul_cert_key}}",
        "verify_incoming": true,
        "verify_outgoing": true
      },
     "internal_rpc":{
         "verify_server_hostname": true
     }
    },
     "telemetry": {
       "disable_hostname" : true
     }
}

{
    "node_name": "{{host_name}}",
    "datacenter": "{{data_center}}",
    "data_dir": "{{consul_data_dir}}",
	"ports": {"https":8501},
	"bind_addr": "0.0.0.0",
    "advertise_addr": "{{host_ip}}",
    "auto_encrypt": {
        "tls": true
    },
    "retry_join":{{masters | to_json }},
    "encrypt": "{{consul_encrypt_key}}",
    "node_meta": {
        "node_type": "worker"
    },
    "ports": {
        "http": -1,
        "grpc" : 8502,
        "https": 8501
    },
    "connect": {
        "enabled": true
    },
    "tls":{
      "defaults":{
        "ca_file": "{{cluster_intermediate_ca_bundle}}",
        "cert_file": "{{consul_cert}}",
        "key_file": "{{consul_cert_key}}",
        "verify_incoming": false,
        "verify_outgoing": true
      },
      "internal_rpc":{
         "verify_server_hostname": true
      }
    },
    "telemetry": {
      "disable_hostname" : true
    }
}
´´´

**Comment by tgross** [comment]
Hi folks, we've got a patch landed for this and it will ship in Nomad 1.4.3, which is coming out shortly.
